src/components/multiSelect/NetworkMultiSelectBottomSheet.tsx

```diff
@@ -0,0 +1,57 @@
+import GorhomBottomSheet from '@gorhom/bottom-sheet'
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import MultiSelectBottomSheet, {
+  ItemProps,
+} from 'src/components/multiSelect/MultiSelectBottomSheet'
+import { useSelector } from 'src/redux/hooks'
+import { NETWORK_NAMES } from 'src/shared/conts'
+import { networksIconSelector } from 'src/tokens/selectors'
+import { NetworkId } from 'src/transactions/types'
+
+interface Props {
+  forwardedRef: React.RefObject<GorhomBottomSheet>
+  onClose?: () => void
+  onOpen?: () => void
+  selectedNetworkIds: Record<NetworkId, boolean>
+  setSelectedNetworkIds: (selectedNetworkIds: Record<NetworkId, boolean>) => void
+}
+
+function NetworkMultiSelectBottomSheet({
+  forwardedRef,
+  onClose,
+  onOpen,
+  selectedNetworkIds,
+  setSelectedNetworkIds,
+}: Props) {
+  const { t } = useTranslation()
+  const allNetworkIds = Object.keys(selectedNetworkIds) as NetworkId[]
+
+  const networkIconByNetworkId = useSelector((state) => networksIconSelector(state, allNetworkIds))
+
+  const textAndIconMap = allNetworkIds.reduce(
+    (acc, networkId) => {
+      acc[networkId] = {
+        text: NETWORK_NAMES[networkId],
+        iconUrl: networkIconByNetworkId[networkId],
+      }
+      return acc
+    },
+    {} as Record<NetworkId, Omit<ItemProps, 'onPress' | 'isSelected'>>
```

Comment:
this same Omit is done in a couple of places, maybe define a type for Item and then have ItemProps extend it?